File: source/static/application/historical.mjs
fetch(`/api/historical/${location.search.split('=')[1]}/`).then(async response => {
    let i = 0
    const json = await response.json()
    const table = document.body.querySelector('table tbody')
    for (let i = 1; i <= json.historical.length; i++) {
        let historic = json.historical[json.historical.length - i]
        let date = new Date(historic.date)
        let tr = document.createElement('tr')
        tr.innerHTML = `
            <td>${json.historical.length - i}</td>
            <td>${date.toLocaleDateString('PT-br', {
                weekday: 'long',
                year: 'numeric',
                month: 'long',
                day: 'numeric',
                // hour: '2-digit',
                // minute: '2-digit',
            })}</td>
            <td>${historic.open}</td>
            <td>${historic.low}</td>
            <td>${historic.high}</td>
            <td>${historic.close}</td>
            <td>${historic.volume}</td>
        `
        table.append(tr)
    }
})

File: source/serializers/historical.py
from .serializer import Serializer
from source.entities.historic import Historic

class Historical(Serializer):

    def __init__(self, entities: list[Historic]):
        self._entities = entities
        super().__init__()

    def handle(self):
        list = []
        for entity in self._entities:
            timeline = entity.timeline
            list.append({
                'id': entity.id,
                'date': self.serialize(timeline.datetime),
                'stock_id': timeline.stock.id,
                'low': entity.low,
                'high': entity.high,
                'open': entity.open,
                'close': entity.close,
                'volume': entity.volume,
            })
        return {
            'historical': list,
        }